dspace-api/src/main/java/org/dspace/content/authority/ChoiceAuthorityServiceImpl.java

```diff
@@ -206,37 +242,93 @@ protected String makeFieldKey(String schema, String element, String qualifier)
-            while(keyIterator.hasNext())
-            {
-                String key = keyIterator.next();
-                String fkey = config2fkey(key.substring(CHOICES_PLUGIN_PREFIX.length()));
-                if (fkey == null)
-                {
-                    log.warn("Skipping invalid ChoiceAuthority configuration property: "+key+": does not have schema.element.qualifier");
-                    continue;
-                }
-
-                // XXX FIXME maybe add sanity check, call
-                // MetadataField.findByElement to make sure it's a real field.
-                ChoiceAuthority ma = (ChoiceAuthority)
-                    pluginService.getNamedPlugin(ChoiceAuthority.class, configurationService.getProperty(key));
-                if (ma == null)
-                {
-                    log.warn("Skipping invalid configuration for "+key+" because named plugin not found: "+configurationService.getProperty(key));
-                    continue;
-                }
-                controller.put(fkey, ma);
-
-                log.debug("Choice Control: For field="+fkey+", Plugin="+ma);
-            }
+            loadChoiceAuthorityConfigurations();
         }
 
         return controller;
     }
 
+	private void loadChoiceAuthorityConfigurations() {
+		// Get all configuration keys starting with a given prefix
+		List<String> propKeys = configurationService.getPropertyKeys(CHOICES_PLUGIN_PREFIX);
+		Iterator<String> keyIterator = propKeys.iterator();
+		while(keyIterator.hasNext())
+		{
+		    String key = keyIterator.next();
+		    String fkey = config2fkey(key.substring(CHOICES_PLUGIN_PREFIX.length()));
+		    if (fkey == null)
+		    {
+		        log.warn("Skipping invalid ChoiceAuthority configuration property: "+key+": does not have schema.element.qualifier");
+		        continue;
+		    }
+
+		    // XXX FIXME maybe add sanity check, call
+		    // MetadataField.findByElement to make sure it's a real field.
+		    String authorityName = configurationService.getProperty(key);
+			ChoiceAuthority ma = (ChoiceAuthority)
+		        pluginService.getNamedPlugin(ChoiceAuthority.class, authorityName);
+		    if (ma == null)
+		    {
+		        log.warn("Skipping invalid configuration for "+key+" because named plugin not found: "+authorityName);
+		        continue;
+		    }
+		    if(!authorities.containsKey(authorityName)) {
+		    	controller.put(fkey, ma);
+		    	authorityNames.add(authorityName);
+		    	authorities.put(authorityName, fkey);
+		    }
+		    else {
+		    	log.warn("Skipping invalid configuration for "+key+" because plugin is alredy in use: "+authorityName +" used by " + authorities.get(authorityName));
+		        continue;		    	
+		    }
+		    
+		    log.debug("Choice Control: For field="+fkey+", Plugin="+ma);
+		}
+		try {
+			DCInputsReader dcInputsReader = new DCInputsReader();
+			for (DCInputSet dcinputSet : dcInputsReader.getAllInputs(Integer.MAX_VALUE, 0)) {
```

Comment:
I understand what you try to do, but I think its better to create a new method for that.

Reply:
If you want more readable the code ok I can encapsulate into a new method. But from now all valuepairs and vocabulary are threated like the others authority plugin and we have to autoregister in the same manner at same place.